src/main/java/com/kodilla/ecommercee/controller/OrderController.java

```diff
@@ -35,22 +49,48 @@ public OrderDto getOrder(@RequestParam long id) throws OrderNotFoundException {
     }
 
     @PostMapping("createOrder")
-    public void createOrder(@RequestBody OrderDto orderDto) throws OrderNotFoundException{
+
+    public void createOrder(@RequestBody OrderDto orderDto) throws OrderNotFoundException, UserNotFoundException {
         Order order = orderService.saveOrder(orderMapper.mapToOrder(orderDto));
-        order.setTrelloCardId(trelloClient.addOrderToList(order.getId(),TrelloClient.NEW_ORDER_LIST).getListId());
+        order.setTrelloCardId(trelloClient.addOrderToList(order.getId(), TrelloClient.NEW_ORDER_LIST).getListId());
     }
 
     @PutMapping("editOrder")
-    public OrderDto editOrder(@RequestBody OrderDto orderDto) throws OrderNotFoundException{
+    public OrderDto editOrder(@RequestBody OrderDto orderDto) throws OrderNotFoundException, UserNotFoundException {
         Order order = orderService.updateOrder(orderMapper.mapToOrder(orderDto));
         trelloClient.updateOrder(order.getId());
         return orderMapper.mapToOrderDto(orderService.updateOrder(orderMapper.mapToOrder(orderDto)));
 
     }
 
     @DeleteMapping("deleteOrder")
-    public void deleteOrder(@RequestParam long id) throws OrderNotFoundException{
+    public void deleteOrder(@RequestParam long id) throws OrderNotFoundException {
         orderService.deleteOrder(id);
         trelloClient.deleteOrder(id);
     }
+
+
+    @PostMapping(value = "orderProcessor")
+    public Order orderProcessor(long userId, List<ProductItem> productItems) throws UserNotFoundException,OrderNotFoundException {
```

Comment:
Czy nie lepiej byłoby zrobić żeby funkcja przyjmowała Cart i z koszyk pobierała liste i użytkownika? Wtedy mamy pewność, że jakimś cudem nie zostanie stworzone zamówienie dla jednego użytkownika z listą zakupów innego.

Reply:
dokładnie o tym samym pomyślałem.
pytanie ten orderProcessor to nie powinien być poprostu createOrder?

Reply:
hmm no racja, dzięki za uwage i biore sie za poprawianie.

Reply:
@MarcinNidecki  wydaje mi sie ,że tak , więc poprostu można usunąć endpointa  createOrder i zastąpić go tym. Tutaj jest przedstawiony cały proces zamówienia.